MetadataExtractor/IO/BufferReader.cs

```diff
@@ -21,15 +21,17 @@ public byte GetByte()
 
     public void GetBytes(scoped Span<byte> bytes)
     {
-        Debug.Assert(_position + bytes.Length <= _bytes.Length, "attempted to read past end of data");
+        if (_position + bytes.Length > _bytes.Length)
+            throw new IOException("End of data reached.");
```

Comment:
Once this merges, I'll go through the regression test suite and check for occurrences of this message, so we can output more helpful messages like above with `directory.AddError("Insufficient bytes for PNG pHYs chunk.");`.

Reply:
#395